static/app/components/forms/controls/rangeSlider/index.tsx

```diff
@@ -0,0 +1,145 @@
+import {forwardRef as reactFowardRef, useEffect, useState} from 'react';
+
+import Input from 'sentry/components/forms/controls/input';
+import Slider from 'sentry/components/forms/controls/rangeSlider/slider';
+import SliderAndInputWrapper from 'sentry/components/forms/controls/rangeSlider/sliderAndInputWrapper';
+import SliderLabel from 'sentry/components/forms/controls/rangeSlider/sliderLabel';
+import type SliderProps from 'sentry/components/forms/controls/rangeSlider/sliderProps';
+import {t} from 'sentry/locale';
+import {defined} from 'sentry/utils';
+
+function RangeSlider({
+  value,
+  allowedValues,
+  showCustomInput,
+  name,
+  disabled,
+  placeholder,
+  formatLabel,
+  className,
+  onBlur,
+  onChange,
+  forwardRef,
+  showLabel = true,
+  ...props
+}: SliderProps) {
+  const [sliderValue, setSliderValue] = useState(
+    allowedValues ? allowedValues.indexOf(Number(value || 0)) : value
+  );
+
+  useEffect(() => {
+    updateSliderValue();
+  }, [value]);
+
+  function updateSliderValue() {
+    if (!defined(value)) {
+      return;
+    }
+
+    const newSliderValueIndex = allowedValues?.indexOf(Number(value || 0)) ?? -1;
+
+    // If `allowedValues` is defined, then `sliderValue` represents index to `allowedValues`
+    if (newSliderValueIndex > -1) {
+      setSliderValue(newSliderValueIndex);
+      return;
+    }
+
+    setSliderValue(value);
+  }
+
+  function getActualValue(newSliderValue: SliderProps['value']): SliderProps['value'] {
+    if (!allowedValues) {
+      return newSliderValue;
+    }
+
+    // If `allowedValues` is defined, then `sliderValue` represents index to `allowedValues`
+    return allowedValues[newSliderValue];
+  }
+
+  function handleInput(e: React.ChangeEvent<HTMLInputElement>) {
+    const newSliderValue = parseInt(e.target.value, 10);
```

Comment:
Currently, the slider will not work if you pass floating point values as parseInt will drop any floats. You can fix this by just changing all parseInt to parseFloat, it acts in exactly the same way as parseInt when ints are passed.

Reply:
This is existing code unrelated to the exports. Moving to new PR https://github.com/getsentry/sentry/pull/33219